compiler/typecheck/TcMType.hs

```diff
@@ -2146,10 +2146,17 @@ tidySigSkol env cx ty tv_prs
         (env', tv') = tidy_tv_bndr env tv
 
     tidy_ty env (FunTy w arg res)
-      = FunTy w (tidyType env arg) (tidy_ty env res)
+      = FunTy (tidyMult env w) (tidyType env arg) (tidy_ty env res)
```

Comment:
So it _was_ a bug after all?

Didn't you fix just that earlier?

Reply:
In #253, I fixed a different function `tidyType`. The pattern is the same, I don't have an example how `tidyType` could do something wrong, but changing `tidySigSkol` resolves #176.

Reply:
I see.